src/test/java/com/spencerwi/either/EitherTest.java

```diff
@@ -286,11 +287,58 @@ boolean isValid() {
         }
     }
 
-    public static class WrapperAroundBoolean {
+    @Nested
+    @DisplayName("Either.Left")
+    public class EitherOrElseTest {
+
+        @Test
+        void orElseOnLeft() {
+            Either<String, Integer> leftEither = Either.left("foo");
+
+            assertThat(leftEither.getLeftOrElse("bar")).isEqualTo("foo");
+            assertThat(leftEither.getLeftOrElse(() -> "bar")).isEqualTo("foo");
+            assertThat(leftEither.getLeftOrElseThrow(i -> new RuntimeException(String.valueOf(i)))).isEqualTo("foo");
+            assertThat(leftEither.getLeftOrElseThrow(() -> new RuntimeException("Int not accepted"))).isEqualTo("foo");
+
+            assertThat(leftEither.getRightOrElse(2)).isEqualTo(2);
+            assertThat(leftEither.getRightOrElse(() -> 2)).isEqualTo(2);
+            assertThatThrownBy(() -> leftEither.getRightOrElseThrow(i -> new RuntimeException(String.valueOf(i))))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessage("foo");
+
+            assertThatThrownBy(() -> leftEither.getRightOrElseThrow(() -> new RuntimeException("String not accepted")))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessage("String not accepted");
+        }
+
+        @Test
+        void orElseOnRight() {
+            Either<String, Integer> rightEither = Either.right(1);
+
+            assertThat(rightEither.getLeftOrElse("bar")).isEqualTo("bar");
+            assertThat(rightEither.getLeftOrElse(() -> "bar")).isEqualTo("bar");
+
+            assertThatThrownBy(() -> rightEither.getLeftOrElseThrow(i -> new RuntimeException(String.valueOf(i))))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessage("1");
+            assertThatThrownBy(() -> rightEither.getLeftOrElseThrow(() -> new RuntimeException("Int not accepted")))
+                    .isInstanceOf(RuntimeException.class)
+                    .hasMessage("Int is not accepted");
+
+            assertThat(rightEither.getRightOrElse(2)).isEqualTo(1);
+            assertThat(rightEither.getRightOrElse(() -> 2)).isEqualTo(1);
+            assertThat(rightEither.getRightOrElseThrow(i -> new RuntimeException(String.valueOf(i)))).isEqualTo(1);
+
+            assertThat(rightEither.getRightOrElseThrow(() -> new RuntimeException("String not accepted"))).isEqualTo(1);
+        }
+    }
```

Comment:
I'd actually prefer that the tests for each of the `orElse` family be separated; that way, if an assertion on line 318 fails, the test can still exercise the assertion on line 320.